app/livechat/imports/server/rest/sms.js

```diff
@@ -77,30 +86,57 @@ API.v1.addRoute('livechat/sms-incoming/:service', {
 			},
 		};
 
-		sendMessage.message.attachments = sms.media.map((curr) => {
+		let file;
+		const attachments = Promise.await(Promise.all(sms.media.map(async (curr) => {
+			const { url: twilioUrl, contentType } = curr;
+
+			if (!file) {
+				const details = {
+					name: 'Upload File',
+					type: contentType,
+					rid,
+					visitorToken: token,
+				};
+
+				const upload = await getUploadFile(details, twilioUrl);
+				file = Object.assign({}, { _id: upload._id, name: upload.name, type: upload.type });
+			}
+
+			const url = FileUpload.getPath(`${ file._id }/${ encodeURI(file.name) }`);
+
 			const attachment = {
-				message_link: curr.url,
+				message_link: url,
 			};
 
-			const { contentType } = curr;
 			switch (contentType.substr(0, contentType.indexOf('/'))) {
 				case 'image':
-					attachment.image_url = curr.url;
+					attachment.image_url = url;
 					break;
 				case 'video':
-					attachment.video_url = curr.url;
+					attachment.video_url = url;
 					break;
 				case 'audio':
-					attachment.audio_url = curr.url;
+					attachment.audio_url = url;
 					break;
 			}
 
 			return attachment;
-		});
+		})));
+
+		const message = {
+			_id: Random.id(),
+			rid: (room && room._id) || Random.id(),
+			token,
+			msg: sms.body,
+			...location && { location },
+			...attachments && { attachments },
+			...file && { file },
+		};
 
-		try {
-			const message = SMSService.response.call(this, Promise.await(Livechat.sendMessage(sendMessage)));
+		Object.assign(sendMessage, { message });
```

Comment:
why not simply?
```suggestion
		sendMessage.message = message;
```

don't even need a variable actually, it could be done like:
```
sendMessage.message = {
	_id: Random.id(),
	rid,
	token,
	msg: sms.body,
	...location && { location },
	...attachments && { attachments },
	...file && { file },
};